docs/HOL_Deploy_Private_Docker_Registry/README.md

```diff
@@ -0,0 +1,447 @@
+
+![](<media/builddefconfig.png>)
+
+ **Step 11.** Add a build step to build the **Clients** Docker image. Click on the **Build** tab, click **Add build step...**, and then click the **Add** button next to the Docker task. Docker task enables you to build, run, push Docker images or execute other operations offered by the Docker CLI.
+
+![](<media/addstepforweb.png>)
+
+Click the pencil icon to enter your preferred task name.
+
+![](<media/edittaskname.png>)
+
+Configure the task (e.g., Build Clients Image) as follows:
+
+![](<media/configbuildclients.png>)
+
+* **Docker Registry Connection**: Select the Docker registry endpoint created earlier.
+* **Action**: Select **Build an image**.
+* **Docker File**: Select the path to Docker file for your **Clients** Component.
+* **Use Default Build Context**: Tick this checkbox. Set the build context to the directory that contains the Docker file.  
+* **Image Name**: Enter your preferred image name (e.g., **clients:$(Build.BuildId)**).
+* **Qualify Image Name**: Tick this checkbox. Qualify the image name with the Docker registry connection's hostname.
+
+**Step 12.** Add a build step to run the **Clients** Docker image. Configure the Docker task (e.g., Run Clients Image) as follows:
+
+![](<media/runclientimage.png>)
+
+* **Docker Registry Connection**: Select the Docker registry endpoint created earlier.
+* **Action**: Select **Run an image**.
+* **Image Name**: Enter the image name (e.g., **clients:$(Build.BuildId)**) you wish to run.
+* **Qualify Image Name**: Tick this checkbox. Qualify the image name with the Docker registry connection's hostname.
+* **Container Name**: Enter your preferred container name (e.g., clients).
+* **Ports**: Enter **80:8080**. Ports in the Docker container to publish to the host.
+
+**Step 13.** Add a build step to inspect the running Container using [Docker Inspect](https://docs.docker.com/engine/reference/commandline/inspect/). Configure the Docker task (e.g., Inspect Clients Container) as follows:
+
+![](<media/inspectclientscontainer.png>)
+
+* **Action**: Select **Run a Docker command**.
+* **Command**: Enter the following command line, and replace the container-name with your container name (e.g., clients):  
+
+    ```
+    inspect <container-name>
+    ```
+
+**Step 14.** Add a build step to scan security vulnerabilities using [Docker Bench for Security](https://github.com/docker/docker-bench-security). Configure the Docker task (e.g., Scan Security Vulnerabilities for Images and Containers) as follows:
+
+![](<media/scansecurity.png>)
+
+* **Action**: Select **Run a Docker command**.
+* **Command**: Enter the following command line:  
+
+    ```
+    run --net host --pid host --cap-add audit_control -v /var/lib:/var/lib -v /var/run/docker.sock:/var/run/docker.sock -v /usr/lib/systemd:/usr/lib/systemd -v /etc:/etc --label docker_bench_security docker/docker-bench-security
```

Comment:
The problem here, is, at each build it will create a new container to run the docker-bench-security... Find a way to re-use this same container or remove it at the end of each build.